src/main/java/uk/nhs/ctp/repos/CaseRepository.java

```diff
@@ -3,26 +3,28 @@
 import java.util.Date;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
-import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 import uk.nhs.ctp.entities.Cases;
 import uk.nhs.ctp.service.search.AuditSearchResultDTO;
 
 @Repository
-public interface CaseRepository extends JpaRepository<Cases, Long> {
+public interface CaseRepository extends PartitionedRepository<Cases, Long> {
   
   @Query("SELECT NEW uk.nhs.ctp.service.search.AuditSearchResultDTO(c.id, c.firstName, c.lastName, c.createdDate) " +
           "FROM Cases c " +
           "WHERE (:from IS NULL OR createdDate > :from) " +
           "AND (:to IS NULL OR createdDate < :to) " +
           "AND (:includeClosed = TRUE OR closedDate IS NULL)" +
-          "AND (:includeIncomplete = TRUE OR triageComplete = TRUE)")
+          "AND (:includeIncomplete = TRUE OR triageComplete = TRUE)" +
+          "AND (:supplierId IS NULL OR supplierId = :supplierId)")
```

Comment:
Does this need the IS NULL clause? I don't think we ever want passing null for the supplier ID to mean all records.